proxy/http2/Http2ClientSession.cc

```diff
@@ -116,9 +116,16 @@ Http2ClientSession::new_connection(NetVConnection *new_vc, MIOBuffer *iobuf, IOB
   this->_read_buffer_reader     = reader ? reader : this->read_buffer->alloc_reader();
 
   // This block size is the buffer size that we pass to SSLWriteBuffer
-  auto buffer_block_size_index   = iobuffer_size_to_index(Http2::write_buffer_block_size, MAX_BUFFER_SIZE_INDEX);
-  this->write_buffer             = new_MIOBuffer(buffer_block_size_index);
-  this->write_buffer->water_mark = Http2::buffer_water_mark;
+  auto buffer_block_size_index = iobuffer_size_to_index(Http2::write_buffer_block_size, MAX_BUFFER_SIZE_INDEX);
+  this->write_buffer           = new_MIOBuffer(buffer_block_size_index);
+
+  uint32_t buffer_water_mark;
+  if (this->_vc->options.http2_buffer_water_mark != 0) {
```

Comment:
Can you help me think through this: is this OK if the value is the default of `-1`? Or is `_vc->options.http2_buffer_water_mark` guaranteed to not be `-1` here? I might be missing something...

Reply:
That's a good callout. I might've posted this PR in mid transition from the PR this required.

The desire is to only apply the change if the override(via `sni.yaml`)has been explicitly been set. Or if it's not explicitly set , to use the default from `records.config`. 

Maybe I should use `std::optional` here as well instead of trying to track/remember sentinel values.